src/app/WriteClient.cpp

```diff
@@ -244,12 +244,21 @@ CHIP_ERROR WriteClient::SendWriteRequest(SessionHandle session, System::Clock::T
     // Create a new exchange context.
     mpExchangeCtx = mpExchangeMgr->NewContext(session, this);
     VerifyOrExit(mpExchangeCtx != nullptr, err = CHIP_ERROR_NO_MEMORY);
-    mpExchangeCtx->SetResponseTimeout(timeout);
+    if (session.IsGroupSession())
+    {
+        // Exchange will be close by WriteClientHandle::SendWriteRequest for group messages
```

Comment:
```suggestion
        // Exchange will be closed by WriteClientHandle::SendWriteRequest for group messages
```
If this stays around, of course.

Reply:
fixed